app/src/main/java/com/kabouzeid/gramophone/model/lyrics/Lyrics.java

```diff
@@ -0,0 +1,49 @@
+package com.kabouzeid.gramophone.model.lyrics;
+
+import com.kabouzeid.gramophone.model.Song;
+
+public class Lyrics {
+    public Song song;
+    public String data;
+
+    boolean parsed = false;
+    boolean valid = false;
+
+    public Lyrics(Song song, String data) {
+        this.song = song;
+        this.data = data;
+    }
+
+    public static Lyrics parse(Song song, String data) {
```

Comment:
@tkashkin I think a [`ServiceLoader`](http://docs.oracle.com/javase/6/docs/api/java/util/ServiceLoader.html) is what you're looking for.